services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractRegistrationService.java

```diff
@@ -110,15 +111,17 @@ public void setTenantInformationService(final TenantInformationService tenantInf
      * {@link #processAssertRegistration(DeviceKey, Span) processAssertRegistration} method to work.
      */
     @Override
-    public Future<RegistrationResult>  assertRegistration(final String tenantId, final String deviceId, final Span span) {
+    public Future<RegistrationResult> assertRegistration(final String tenantId, final String deviceId, final Span span) {
 
         Objects.requireNonNull(tenantId);
         Objects.requireNonNull(deviceId);
         Objects.requireNonNull(span);
 
         return this.tenantInformationService
                 .tenantExists(tenantId, span)
-                .flatMap(tenantKey -> processAssertRegistration(DeviceKey.from(tenantKey, deviceId), span)
+                .flatMap(tenantKeyResult -> tenantKeyResult.isError()
```

Comment:
is there a particular reason why you are using *flatMap* in some places and *compose* in other places?

Reply:
Changed everything to `compose` to make it uniform across the codebase.